packages/app/src/components/catalog/EntityPage/EntityPage.tsx

```diff
@@ -0,0 +1,143 @@
+import { dynamicEntityTab, DynamicEntityTabProps } from './DynamicEntityTab';
+
+/**
+ * Displays the tabs and content for a catalog entity
+ * *Note:* do not convert convert this to a component or wrap the return value
+ * @param entityTabOverrides
+ * @returns
+ */
+export const entityPage = (
+  entityTabOverrides: Record<
+    string,
+    Omit<DynamicEntityTabProps, 'path' | 'if' | 'children'>
+  > = {},
+) => {
+  const tabRules: Record<
+    string,
+    Omit<DynamicEntityTabProps, 'path' | 'title' | 'mountPoint' | 'children'>
+  > = {
+    '/api': {
+      if: (entity: Entity) =>
+        isType('service')(entity) && isKind('component')(entity),
+    },
+    '/dependencies': {
+      if: isKind('component'),
+    },
+    '/definition': {
+      if: isKind('api'),
+    },
+    '/system': {
+      if: isKind('system'),
+    },
+  };
+
+  const tabChildren: Record<
+    string,
+    Omit<DynamicEntityTabProps, 'path' | 'title' | 'mountPoint' | 'if'>
+  > = {
+    '/': {
+      children: <OverviewTabContent />,
+    },
+    '/api': {
+      children: <ApiTabContent />,
+    },
+    '/dependencies': {
+      children: <DependenciesTabContent />,
+    },
+    '/definition': {
+      children: <DefinitionTabContent />,
+    },
+    '/system': {
+      children: <DiagramTabContent />,
+    },
+  };
+
+  const defaultTabs: Record<
+    string,
+    Omit<DynamicEntityTabProps, 'if' | 'children' | 'path'>
+  > = {
+    '/': {
+      title: 'Overview',
+      mountPoint: 'entity.page.overview',
+    },
+    '/topology': {
+      title: 'Topology',
+      mountPoint: 'entity.page.topology',
+    },
+    '/issues': {
+      title: 'Issues',
+      mountPoint: 'entity.page.issues',
+    },
+    '/pr': {
+      title: 'Pull/Merge Requests',
+      mountPoint: 'entity.page.pull-requests',
+    },
+    '/ci': {
+      title: 'CI',
+      mountPoint: 'entity.page.ci',
+    },
+    '/cd': {
+      title: 'CD',
+      mountPoint: 'entity.page.cd',
+    },
+    '/kubernetes': {
+      title: 'Kubernetes',
+      mountPoint: 'entity.page.kubernetes',
+    },
+    '/image-registry': {
+      title: 'Image Registry',
+      mountPoint: 'entity.page.image-registry',
+    },
+    '/monitoring': {
+      title: 'Monitoring',
+      mountPoint: 'entity.page.monitoring',
+    },
+    '/lighthouse': {
+      title: 'Lighthouse',
+      mountPoint: 'entity.page.lighthouse',
+    },
+    '/api': {
+      title: 'Api',
+      mountPoint: 'entity.page.api',
+    },
+    '/dependencies': {
+      title: 'Dependencies',
+      mountPoint: 'entity.page.dependencies',
+    },
+    '/docs': {
+      title: 'Docs',
+      mountPoint: 'entity.page.docs',
+    },
+    '/definition': {
+      title: 'Definition',
+      mountPoint: 'entity.page.definition',
+    },
+    '/system': {
+      title: 'Diagram',
+      mountPoint: 'entity.page.diagram',
+    },
+  };
+  return (
+    <EntityLayout>
+      {Object.entries({ ...defaultTabs, ...entityTabOverrides }).map(
```

Comment:
@gashcrumb Is there a way to configure the order of the tab? For example if I wanted to add a tab just after `Topology` one or just before `CI` one?

Reply:
We could probably add something like `insertBefore` and `insertAfter` in the extension to let the users configure ordering of tabs. We did something similar in OpenShift Console Dynamic Plugin extension points.

Reply:
Yep, we certainly could do that, either now or in a subsequent improvement.  It's sort of possible to influence the order of the tabs with this change by nature of the way the dictionaries are combined, but having a more explicit means to do this would ultimately be nicer.

Reply:
Thinking on this further, with this change you can specify whatever order you like by overriding the default list in the config.  If in the future we decide to reduce [this list](https://github.com/janus-idp/backstage-showcase/pull/1173/files#diff-dc23b2c0e4792453496e74f891819cf0cf5172dd3d6b426bca6b9f85aef4d7e5R64-R128) and move the tab config into the plugin config (for example, move `/topology` to an `entityTabs` setting in [here](https://github.com/janus-idp/backstage-showcase/blob/main/dynamic-plugins.default.yaml#L229)) then we would definitely need these additional directives to influence the tab order.

Reply:
For setting up a custom order right now users would need to override the complete list of default tabs, right? Otherwise I am not sure if `{ ...defaultTabs, ...entityTabOverrides }` would result in the desired order every time.

Reply:
The order is the order of enumerated properties of the combined object which indeed is not necessarily straightforward to reason through.  But keep in mind that the original goal of this effort is mainly to allow users to specify new tabs in addition to what is provided out of the box so that a user's plugin could have it's own entity page.  Being able to customize the existing tabs is also nice but not a primary goal of this feature.